In [1]:
import pandas as pd
import numpy as py
health_expenditure = pd.read_csv (r'/Users/abdullahwaraich/Downloads/healthexpenditurebyareaandsource.csv')
health_expenditure


financial_year state  area_of_expenditure broad_source_of_funding  \
0           1997-98   NSW       Administration              Government   
1           1997-98   NSW       Administration              Government   
2           1997-98   NSW       Administration          Non-government   
3           1997-98   NSW  Aids and appliances              Government   
4           1997-98   NSW  Aids and appliances          Non-government   
...             ...   ...                  ...                     ...   
7095            NaN   NaN                  NaN                     NaN   
7096            NaN   NaN                  NaN                     NaN   
7097            NaN   NaN                  NaN                     NaN   
7098            NaN   NaN                  NaN                     NaN   
7099            NaN   NaN                  NaN                     NaN   

          detailed_source_of_funding  real_expenditure_millions  
0              Australian Government                      315.0  
1                    State and local                      120.0  
2     Private health insurance funds                      314.0  
3              Australian Government                       65.0  
4                        Individuals                      168.0  
...                              ...                        ...  
7095                             NaN                        NaN  
7096                             NaN                        NaN  
7097                             NaN                        NaN  
7098                             NaN                        NaN  
7099                             NaN                        NaN  

[7100 rows x 6 columns]

In [2]:
health_expenditure.describe()
health_expenditure_by_state = health_expenditure.groupby(['state']).real_expenditure_millions.sum()
health_expenditure_by_state




state
ACT     28260.0
NSW    482366.0
NT      18915.0
QLD    289676.0
SA     114623.0
TAS     34294.0
VIC    360217.0
WA     149887.0
Name: real_expenditure_millions, dtype: float64

In [3]:
health_expenditure_by_state.sum()
health_expenditure_total = health_expenditure_by_state.sum()

Summarised above is the total expenditure for health by all states from 1997 to 2012 (financial years)
Now, the goal is to determine what each state spent the most on and whether or not it was government or non government funding. So basically just looking for area of expenditure and where the funding is coming from. 

In [4]:
health_expenditure_funding = health_expenditure.groupby(['state', 'area_of_expenditure', 'broad_source_of_funding'])
health_expenditure_funding.head()



financial_year state  area_of_expenditure broad_source_of_funding  \
0           1997-98   NSW       Administration              Government   
1           1997-98   NSW       Administration              Government   
2           1997-98   NSW       Administration          Non-government   
3           1997-98   NSW  Aids and appliances              Government   
4           1997-98   NSW  Aids and appliances          Non-government   
...             ...   ...                  ...                     ...   
6778            NaN   NaN                  NaN                     NaN   
6779            NaN   NaN                  NaN                     NaN   
6780            NaN   NaN                  NaN                     NaN   
6781            NaN   NaN                  NaN                     NaN   
6782            NaN   NaN                  NaN                     NaN   

          detailed_source_of_funding  real_expenditure_millions  
0              Australian Government                      315.0  
1                    State and local                      120.0  
2     Private health insurance funds                      314.0  
3              Australian Government                       65.0  
4                        Individuals                      168.0  
...                              ...                        ...  
6778                             NaN                        NaN  
6779                             NaN                        NaN  
6780                             NaN                        NaN  
6781                             NaN                        NaN  
6782                             NaN                        NaN  

[1205 rows x 6 columns]

In [5]:
health_expenditure_areas = health_expenditure.groupby(['state','area_of_expenditure']).real_expenditure_millions.sum()
health_expenditure_areas['VIC']

area_of_expenditure
Administration                    7935.0
Aids and appliances              10298.0
All other medications            18573.0
Benefit-paid pharmaceuticals     25903.0
Capital expenditure              14903.0
Community health                 11702.0
Dental services                  26996.0
Medical expense tax rebate           0.0
Medical services                 66086.0
Other health practitioners       15492.0
Patient transport services        6380.0
Private hospitals                30321.0
Public health                     6278.0
Public hospitals                106195.0
Research                         13155.0
Name: real_expenditure_millions, dtype: float64

Above is some code that will allow you to determine the spending (in millions) for different categories (areas) of spending. 

In [6]:
health_expenditure_areas = health_expenditure.groupby(['state','detailed_source_of_funding']).real_expenditure_millions.sum()
health_expenditure_vic = health_expenditure_areas['VIC'].sum()
health_expenditure_vic


360217.0

This is the amount, in millions, that VIC spent on health in the designated period. Although, the Public Hospitals Expenditure seems a bit suspicious given that the other ones are relatively close to it. 

In [7]:
a = int(health_expenditure_vic)
b = int(health_expenditure_total)
c = round (a/b,4)
print(c)
print(str(c*100)+'%')


0.2437
24.37%


The above suggests that VIC makes up 24.37% of all government spending on Health. The next step is now to save all of this as a function, so that instead of having to calculate all of this individually, it can all be calculated by calling a single function, eg. calling spending_by('NSW') to determine the percentage of government spending on health by NSW. 
The required features are that: 
a. having health_expenditure_areas (as we are targetting detailed sources of funding at the moment) and then calculating its sum. 
b. Dividing this sum by the total, and from there calculating the % of all government spending. 
Let's see how this goes!

In [8]:
#I will use this to store the actual function and the code below to see if it works.

def spending_by(state): 
    health_expenditure_areas = health_expenditure.groupby(['state','detailed_source_of_funding']).real_expenditure_millions.sum()
    health_expenditure_state = health_expenditure_areas[str(state)].sum()
    health_expenditure_state
    a = int(health_expenditure_state)
    b = int(health_expenditure_total)
    c = round (a/b,4)
    print(c)
    print(str(c*100)+'%')
    


In [9]:
state = input('What state do you want?') #This will ask for an input (what state?), from there it let's you calculate what percentage of spending is by that particular state.
spending_by(state) 

0.3263
32.629999999999995%


So, this tells us that the total percentage of all government spending on Health was 32.62999....% by NSW. This seems quite reasonable, given that NSW has the largest population. We can also use the same function to calculate how much spending by other states. But the fact that we define the variable inside the function can spell trouble, so we will need to fix that later. 

In [10]:
health_expenditure_areas = health_expenditure.groupby(['state','detailed_source_of_funding']).real_expenditure_millions.sum()
health_expenditure_state = health_expenditure_areas['VIC'].sum()   #This is for VIC. 
health_expenditure_state
a = int(health_expenditure_state)
b = int(health_expenditure_total)
c = round (a/b,4)
print(c)
print(str(c*100)+'%')
#Now, we don't really need to use this much code to calculate the percentages, we can just use the function that was defined earlier.

0.2437
24.37%
